lib/developer_portal/app/controllers/developer_portal/admin/applications_controller.rb

```diff
@@ -172,11 +172,12 @@ def accepted_application_params
     # cinstance[*] naming is present for legacy reasons
     application_attributes = params[:application] || params[:cinstance]
     return {} unless application_attributes
+
     permitted_params = fields_definitions + %i[plan_id redirect_url]
     application_attributes.permit(*permitted_params)
   end
 
   def fields_definitions
-    FieldsDefinition.by_provider(site_account).by_target('Cinstance').pluck(:name)
+    FieldsDefinition.by_provider(site_account).by_target('Cinstance').reject(&:read_only).map(&:name)
```

Comment:
@mayorova I applied your suggestions. But I didn't update this controller because I think it's not a good fit for the new method.

@akostadinov I think your comments are more or less aligned to what Daria told me. Basically, extract the read-only filtering logic to it's own method. Please take a look.

Reply:
Well, a further suggestion would be to:
1. add a `read_only` scope for fields definition, like:
```
  scope :read_only, ->(read_only = true) { where(read_only: read_only) }
```

so, we could replace this with 
```
FieldsDefinition.by_provider(site_account).by_target('Cinstance').read_only(false).pluck(:name)
```
I think this will save us from two array transformations, and improve readability a bit.